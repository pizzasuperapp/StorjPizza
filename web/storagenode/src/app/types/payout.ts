// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import {
    EstimatedPayout,
    PayoutPeriod,
    SatelliteHeldHistory,
    SatellitePayoutForPeriod,
    TotalPayments,
    TotalPaystubForPeriod,
} from '@/storagenode/payouts/payouts';

/**
 * Holds 'start' and 'end' of payout period range.
 */
export class PayoutInfoRange {
    public constructor(
        public start: PayoutPeriod | null = null,
        public end: PayoutPeriod = new PayoutPeriod(),
    ) {}
}

/**
 * Holds all payout module state.
 */
export class PayoutState {
    public constructor (
        public totalPaystubForPeriod: TotalPaystubForPeriod = new TotalPaystubForPeriod(),
        public periodRange: PayoutInfoRange = new PayoutInfoRange(),
        public totalPayments: TotalPayments = new TotalPayments(),
        public currentMonthEarnings: number = 0,
        public heldPercentage: number = 0,
        public payoutPeriods: PayoutPeriod[] = [],
        public heldHistory: SatelliteHeldHistory[] = [],
        public payoutHistory: SatellitePayoutForPeriod[] = [],
        public payoutHistoryPeriod: string = '',
        public estimation: EstimatedPayout = new EstimatedPayout(),
        public payoutHistoryAvailablePeriods: PayoutPeriod[] = [],
    ) {}
}

export interface StoredMonthsByYear {
    [key: number]: MonthButton[];
}

/**
 * Holds all months names.
 */
export const monthNames = [
    'January', 'February', 'March', 'April',
    'May', 'June', 'July',	'August',
    'September', 'October', 'November',	'December',
];

/**
 * Describes month button entity for calendar.
 */
export class MonthButton {
    public constructor(
        public year: number = 0,
        public index: number = 0,
        public active: boolean = false,
        public selected: boolean = false,
    ) {}

    /**
     * Returns month label depends on index.
     */
    public get name(): string {
        return monthNames[this.index] ? monthNames[this.index].slice(0, 3) : '';
    }
}
