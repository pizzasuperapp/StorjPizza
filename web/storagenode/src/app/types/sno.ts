// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import {
    BandwidthUsed,
    EgressUsed,
    IngressUsed,
    Node,
    SatelliteInfo,
    SatelliteScores,
    Stamp,
    Utilization,
} from '@/storagenode/sno/sno';

/**
 * Holds all node module state.
 */
export class StorageNodeState {
    public info: Node = new Node();
    public utilization: Utilization = new Utilization();
    public satellites: SatelliteInfo[] = [];
    public disqualifiedSatellites: SatelliteInfo[] = [];
    public suspendedSatellites: SatelliteInfo[] = [];
    public selectedSatellite: SatelliteInfo = new SatelliteInfo();
    public bandwidthChartData: BandwidthUsed[] = [];
    public egressChartData: EgressUsed[] = [];
    public ingressChartData: IngressUsed[] = [];
    public storageChartData: Stamp[] = [];
    public storageSummary = 0;
    public bandwidthSummary = 0;
    public egressSummary = 0;
    public ingressSummary = 0;
    public satellitesScores: SatelliteScores[] = [];
    public audits: SatelliteScores = new SatelliteScores();
}
