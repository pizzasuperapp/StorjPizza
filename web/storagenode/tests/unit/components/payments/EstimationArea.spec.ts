// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import Vuex from 'vuex';

import EstimationArea from '@/app/components/payments/EstimationArea.vue';

import { APPSTATE_MUTATIONS, appStateModule } from '@/app/store/modules/appState';
import { newNodeModule, NODE_MUTATIONS } from '@/app/store/modules/node';
import { newPayoutModule, PAYOUT_MUTATIONS } from '@/app/store/modules/payout';
import { PayoutInfoRange } from '@/app/types/payout';
import { PayoutHttpApi } from '@/storagenode/api/payout';
import { StorageNodeApi } from '@/storagenode/api/storagenode';
import {
    EstimatedPayout,
    PayoutPeriod, Paystub,
    PreviousMonthEstimatedPayout,
    TotalPaystubForPeriod,
} from '@/storagenode/payouts/payouts';
import { PayoutService } from '@/storagenode/payouts/service';
import { StorageNodeService } from '@/storagenode/sno/service';
import { Satellites } from '@/storagenode/sno/sno';
import { createLocalVue, shallowMount } from '@vue/test-utils';

const localVue = createLocalVue();
localVue.use(Vuex);

localVue.filter('centsToDollars', (cents: number): string => {
    return `$${(cents / 100).toFixed(2)}`;
});

const nodeApi = new StorageNodeApi();
const nodeService = new StorageNodeService(nodeApi);
const nodeModule = newNodeModule(nodeService);
const payoutApi = new PayoutHttpApi();
const payoutService = new PayoutService(payoutApi);
const payoutModule = newPayoutModule(payoutService);

const store = new Vuex.Store({ modules: { payoutModule, appStateModule, node: nodeModule }});

describe('EstimationArea', (): void => {
    it('renders correctly with actual values and current period', async (): Promise<void> => {
        const _Date = Date;
        const mockedDate1 = new Date(1580522290000); // Sat Feb 01 2020
        const mockedDate2 = new Date(1577982290000); // Thu Jan 02 2020
        const allSatellitesInfo = new Satellites();
        allSatellitesInfo.joinDate = mockedDate2;
        global.Date = jest.fn(() => mockedDate1);

        const estimatedPayout = new EstimatedPayout(
            new PreviousMonthEstimatedPayout(
                100000000,
                200000000,
                300000000,
                400000000,
                500000000,
                600000000,
                700000000,
                800000000,
                900000000,
            ),
        );

        await store.commit(PAYOUT_MUTATIONS.SET_ESTIMATION, estimatedPayout);
        await store.commit(NODE_MUTATIONS.SELECT_ALL_SATELLITES, allSatellitesInfo);
        await store.commit(PAYOUT_MUTATIONS.SET_RANGE, new PayoutInfoRange(null, new PayoutPeriod(2020, 1)));

        const wrapper = shallowMount(EstimationArea, {
            store,
            localVue,
        });

        expect(wrapper).toMatchSnapshot();

        global.Date = _Date;

        wrapper.destroy();
    });

    it('renders correctly with actual values and current period and not first day on month', async (): Promise<void> => {
        const _Date = Date;
        const mockedDate1 = new Date(1580722290000); // Sat Feb 03 2020
        const mockedDate2 = new Date(1577982290000); // Thu Jan 02 2020
        const allSatellitesInfo = new Satellites();
        allSatellitesInfo.joinDate = mockedDate2;
        global.Date = jest.fn(() => mockedDate1);

        const estimatedPayout = new EstimatedPayout(
            new PreviousMonthEstimatedPayout(
                10000000,
                20000000,
                30000000,
                40000000,
                50000000,
                60000000,
                70000000,
                80000000,
                90000000,
            ),
            new PreviousMonthEstimatedPayout(),
            1200000000,
        );

        await store.commit(PAYOUT_MUTATIONS.SET_ESTIMATION, estimatedPayout);
        await store.commit(NODE_MUTATIONS.SELECT_ALL_SATELLITES, allSatellitesInfo);
        await store.commit(PAYOUT_MUTATIONS.SET_RANGE, new PayoutInfoRange(null, new PayoutPeriod(2020, 1)));

        const wrapper = shallowMount(EstimationArea, {
            store,
            localVue,
        });

        expect(wrapper).toMatchSnapshot();

        global.Date = _Date;

        wrapper.destroy();
    });

    it('renders correctly with actual values and previous period without paystub', async (): Promise<void> => {
        const _Date = Date;
        const mockedDate1 = new Date(1580522290000); // Sat Feb 01 2020
        const mockedDate2 = new Date(1577982290000); // Thu Jan 02 2020
        const allSatellitesInfo = new Satellites();
        allSatellitesInfo.joinDate = mockedDate2;
        global.Date = jest.fn(() => mockedDate1);

        const estimatedPayout = new EstimatedPayout(
            new PreviousMonthEstimatedPayout(),
            new PreviousMonthEstimatedPayout(
                100000000,
                200000000,
                300000000,
                400000000,
                500000000,
                600000000,
                700000000,
                800000000,
                900000000,
            ),
        );

        await store.commit(PAYOUT_MUTATIONS.SET_ESTIMATION, estimatedPayout);
        await store.commit(NODE_MUTATIONS.SELECT_ALL_SATELLITES, allSatellitesInfo);
        await store.commit(PAYOUT_MUTATIONS.SET_RANGE, new PayoutInfoRange(null, new PayoutPeriod(2020, 0)));

        const wrapper = shallowMount(EstimationArea, {
            store,
            localVue,
        });

        expect(wrapper).toMatchSnapshot();

        global.Date = _Date;

        wrapper.destroy();
    });

    it('renders correctly with actual values and historical period with paystub', async (): Promise<void> => {
        const _Date = Date;
        const mockedDate1 = new Date(1580522290000); // Sat Feb 01 2020
        const mockedDate2 = new Date(1577982290000); // Thu Jan 02 2020
        const payoutPeriod = new PayoutPeriod(2020, 0);
        const allSatellitesInfo = new Satellites();
        allSatellitesInfo.joinDate = mockedDate2;
        global.Date = jest.fn(() => mockedDate1);

        const paystub = new Paystub();
        paystub.held = 777777;
        paystub.paid = 555555;
        paystub.surgePercent = 300;
        paystub.distributed = 333333;
        const totalPaystubForPeriod = new TotalPaystubForPeriod([paystub]);

        await store.commit(PAYOUT_MUTATIONS.SET_PERIODS, [payoutPeriod]);
        await store.commit(PAYOUT_MUTATIONS.SET_PAYOUT_INFO, totalPaystubForPeriod);
        await store.commit(NODE_MUTATIONS.SELECT_ALL_SATELLITES, allSatellitesInfo);
        await store.commit(PAYOUT_MUTATIONS.SET_RANGE, new PayoutInfoRange(null, payoutPeriod));

        const wrapper = shallowMount(EstimationArea, {
            store,
            localVue,
        });

        expect(wrapper).toMatchSnapshot();

        wrapper.destroy();

        global.Date = _Date;
    });

    it('renders correctly with actual values and historical period without paystub', async (): Promise<void> => {
        const _Date = Date;
        const mockedDate1 = new Date(1580522290000); // Sat Feb 01 2020
        const mockedDate2 = new Date(1577982290000); // Thu Jan 02 2020
        const payoutPeriod = new PayoutPeriod(2020, 0);
        const allSatellitesInfo = new Satellites();
        allSatellitesInfo.joinDate = mockedDate2;
        global.Date = jest.fn(() => mockedDate1);

        await store.commit(PAYOUT_MUTATIONS.SET_PERIODS, [payoutPeriod]);
        await store.commit(NODE_MUTATIONS.SELECT_ALL_SATELLITES, allSatellitesInfo);
        await store.commit(PAYOUT_MUTATIONS.SET_RANGE, new PayoutInfoRange(null, payoutPeriod));

        await store.commit(APPSTATE_MUTATIONS.SET_NO_PAYOUT_INFO, true);

        const wrapper = shallowMount(EstimationArea, {
            store,
            localVue,
        });

        expect(wrapper).toMatchSnapshot();

        wrapper.destroy();

        global.Date = _Date;
    });
});
