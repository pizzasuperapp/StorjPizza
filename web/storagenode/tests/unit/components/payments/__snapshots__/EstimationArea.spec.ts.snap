// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EstimationArea renders correctly with actual values and current period 1`] = `
<div class="estimation-container">
  <div class="estimation-container__header">
    <p class="estimation-container__header__title">Info &amp; Estimation, <span class="estimation-container__header__period">Feb 2020</span></p>
    <div class="estimation-container__header__selection-area"><button name="Select Current Period" type="button" class="estimation-container__header__selection-area__item active">
        <p class="estimation-container__header__selection-area__item__label long-text">
          Current Period
        </p>
        <p class="estimation-container__header__selection-area__item__label short-text">
          Current Per.
        </p>
      </button>
      <estimationperioddropdown-stub class="estimation-container__header__selection-area__item"></estimationperioddropdown-stub>
    </div>
  </div>
  <div class="estimation-container__divider"></div>
  <div class="estimation-table-container">
    <div class="estimation-table-container__labels-area">
      <div class="column justify-start column-1">
        <p class="estimation-table-container__labels-area__text">Name</p>
      </div>
      <div class="column justify-start column-2">
        <p class="estimation-table-container__labels-area__text">Type</p>
      </div>
      <div class="column justify-start column-3">
        <p class="estimation-table-container__labels-area__text">Price</p>
      </div>
      <div class="column justify-start column-4">
        <p class="estimation-table-container__labels-area__text">Disk</p>
      </div>
      <div class="column justify-start column-5">
        <p class="estimation-table-container__labels-area__text">Bandwidth</p>
      </div>
      <div class="column justify-end column-6">
        <p class="estimation-table-container__labels-area__text">Payout</p>
      </div>
    </div>
    <div class="estimation-table-container__info-area">
      <div class="column justify-start column-1">
        <p class="estimation-table-container__info-area__text">Download</p>
      </div>
      <div class="column justify-start column-2">
        <p class="estimation-table-container__info-area__text">Egress</p>
      </div>
      <div class="column justify-start column-3">
        <p class="estimation-table-container__info-area__text">$20 / TB</p>
      </div>
      <div class="column justify-start column-4">
        <p class="estimation-table-container__info-area__text">--</p>
      </div>
      <div class="column justify-start column-5">
        <p class="estimation-table-container__info-area__text">100MB</p>
      </div>
      <div class="column justify-end column-6">
        <p class="estimation-table-container__info-area__text">$2000000.00</p>
      </div>
    </div>
    <div class="estimation-table-container__info-area">
      <div class="column justify-start column-1">
        <p class="estimation-table-container__info-area__text">Repair &amp; Audit</p>
      </div>
      <div class="column justify-start column-2">
        <p class="estimation-table-container__info-area__text">Egress</p>
      </div>
      <div class="column justify-start column-3">
        <p class="estimation-table-container__info-area__text">$10 / TB</p>
      </div>
      <div class="column justify-start column-4">
        <p class="estimation-table-container__info-area__text">--</p>
      </div>
      <div class="column justify-start column-5">
        <p class="estimation-table-container__info-area__text">300MB</p>
      </div>
      <div class="column justify-end column-6">
        <p class="estimation-table-container__info-area__text">$4000000.00</p>
      </div>
    </div>
    <div class="estimation-table-container__info-area">
      <div class="column justify-start column-1">
        <p class="estimation-table-container__info-area__text">Disk Average Month</p>
      </div>
      <div class="column justify-start column-2">
        <p class="estimation-table-container__info-area__text">Storage</p>
      </div>
      <div class="column justify-start column-3">
        <p class="estimation-table-container__info-area__text">$1.5 / TBm</p>
      </div>
      <div class="column justify-start column-4">
        <p class="estimation-table-container__info-area__text">500MBm</p>
      </div>
      <div class="column justify-start column-5">
        <p class="estimation-table-container__info-area__text">--</p>
      </div>
      <div class="column justify-end column-6">
        <p class="estimation-table-container__info-area__text">$6000000.00</p>
      </div>
    </div>
    <div class="estimation-table-container__info-area">
      <div class="column justify-start column-1">
        <p class="estimation-table-container__info-area__text">Gross Total</p>
      </div>
      <div class="column justify-start column-2"></div>
      <div class="column justify-start column-3"></div>
      <div class="column justify-start column-4"></div>
      <div class="column justify-start column-5"></div>
      <div class="column justify-end column-6">
        <p class="estimation-table-container__info-area__text">$17000000.00</p>
      </div>
    </div>
    <!---->
    <div class="estimation-table-container__held-area">
      <p class="estimation-table-container__held-area__text">Held Back</p>
      <p class="estimation-table-container__held-area__text">-$9000000.00</p>
    </div>
    <!---->
    <div class="estimation-table-container__net-total-area">
      <div class="column justify-start column-1">
        <p class="estimation-table-container__net-total-area__text">NET TOTAL</p>
      </div>
      <div class="column justify-start column-2"></div>
      <div class="column justify-start column-3"></div>
      <div class="column justify-start column-4">
        <p class="estimation-table-container__net-total-area__text">500MBm</p>
      </div>
      <div class="column justify-start column-5">
        <p class="estimation-table-container__net-total-area__text">400MB</p>
      </div>
      <div class="column justify-end column-6">
        <p class="estimation-table-container__net-total-area__text">$8000000.00</p>
      </div>
    </div>
    <!---->
  </div>
  <!---->
  <!---->
</div>
`;

exports[`EstimationArea renders correctly with actual values and current period and not first day on month 1`] = `
<div class="estimation-container">
  <div class="estimation-container__header">
    <p class="estimation-container__header__title">Info &amp; Estimation, <span class="estimation-container__header__period">Feb 2020</span></p>
    <div class="estimation-container__header__selection-area"><button name="Select Current Period" type="button" class="estimation-container__header__selection-area__item active">
        <p class="estimation-container__header__selection-area__item__label long-text">
          Current Period
        </p>
        <p class="estimation-container__header__selection-area__item__label short-text">
          Current Per.
        </p>
      </button>
      <estimationperioddropdown-stub class="estimation-container__header__selection-area__item"></estimationperioddropdown-stub>
    </div>
  </div>
  <div class="estimation-container__divider"></div>
  <div class="estimation-table-container">
    <div class="estimation-table-container__labels-area">
      <div class="column justify-start column-1">
        <p class="estimation-table-container__labels-area__text">Name</p>
      </div>
      <div class="column justify-start column-2">
        <p class="estimation-table-container__labels-area__text">Type</p>
      </div>
      <div class="column justify-start column-3">
        <p class="estimation-table-container__labels-area__text">Price</p>
      </div>
      <div class="column justify-start column-4">
        <p class="estimation-table-container__labels-area__text">Disk</p>
      </div>
      <div class="column justify-start column-5">
        <p class="estimation-table-container__labels-area__text">Bandwidth</p>
      </div>
      <div class="column justify-end column-6">
        <p class="estimation-table-container__labels-area__text">Payout</p>
      </div>
    </div>
    <div class="estimation-table-container__info-area">
      <div class="column justify-start column-1">
        <p class="estimation-table-container__info-area__text">Download</p>
      </div>
      <div class="column justify-start column-2">
        <p class="estimation-table-container__info-area__text">Egress</p>
      </div>
      <div class="column justify-start column-3">
        <p class="estimation-table-container__info-area__text">$20 / TB</p>
      </div>
      <div class="column justify-start column-4">
        <p class="estimation-table-container__info-area__text">--</p>
      </div>
      <div class="column justify-start column-5">
        <p class="estimation-table-container__info-area__text">10MB</p>
      </div>
      <div class="column justify-end column-6">
        <p class="estimation-table-container__info-area__text">$200000.00</p>
      </div>
    </div>
    <div class="estimation-table-container__info-area">
      <div class="column justify-start column-1">
        <p class="estimation-table-container__info-area__text">Repair &amp; Audit</p>
      </div>
      <div class="column justify-start column-2">
        <p class="estimation-table-container__info-area__text">Egress</p>
      </div>
      <div class="column justify-start column-3">
        <p class="estimation-table-container__info-area__text">$10 / TB</p>
      </div>
      <div class="column justify-start column-4">
        <p class="estimation-table-container__info-area__text">--</p>
      </div>
      <div class="column justify-start column-5">
        <p class="estimation-table-container__info-area__text">30MB</p>
      </div>
      <div class="column justify-end column-6">
        <p class="estimation-table-container__info-area__text">$400000.00</p>
      </div>
    </div>
    <div class="estimation-table-container__info-area">
      <div class="column justify-start column-1">
        <p class="estimation-table-container__info-area__text">Disk Average Month</p>
      </div>
      <div class="column justify-start column-2">
        <p class="estimation-table-container__info-area__text">Storage</p>
      </div>
      <div class="column justify-start column-3">
        <p class="estimation-table-container__info-area__text">$1.5 / TBm</p>
      </div>
      <div class="column justify-start column-4">
        <p class="estimation-table-container__info-area__text">50MBm</p>
      </div>
      <div class="column justify-start column-5">
        <p class="estimation-table-container__info-area__text">--</p>
      </div>
      <div class="column justify-end column-6">
        <p class="estimation-table-container__info-area__text">$600000.00</p>
      </div>
    </div>
    <div class="estimation-table-container__info-area">
      <div class="column justify-start column-1">
        <p class="estimation-table-container__info-area__text">Gross Total</p>
      </div>
      <div class="column justify-start column-2"></div>
      <div class="column justify-start column-3"></div>
      <div class="column justify-start column-4"></div>
      <div class="column justify-start column-5"></div>
      <div class="column justify-end column-6">
        <p class="estimation-table-container__info-area__text">$1700000.00</p>
      </div>
    </div>
    <!---->
    <div class="estimation-table-container__held-area">
      <p class="estimation-table-container__held-area__text">Held Back</p>
      <p class="estimation-table-container__held-area__text">-$900000.00</p>
    </div>
    <!---->
    <div class="estimation-table-container__net-total-area">
      <div class="column justify-start column-1">
        <p class="estimation-table-container__net-total-area__text">NET TOTAL</p>
      </div>
      <div class="column justify-start column-2"></div>
      <div class="column justify-start column-3"></div>
      <div class="column justify-start column-4">
        <p class="estimation-table-container__net-total-area__text">50MBm</p>
      </div>
      <div class="column justify-start column-5">
        <p class="estimation-table-container__net-total-area__text">40MB</p>
      </div>
      <div class="column justify-end column-6">
        <p class="estimation-table-container__net-total-area__text">$800000.00</p>
      </div>
    </div>
    <!---->
  </div>
  <div class="estimation-container__payout-area">
    <div class="estimation-container__payout-area__left-area">
      <p class="title-text">Estimated Payout</p>
      <p class="additional-text">At the end of the month if the load keeps the same for the rest of the month.</p>
    </div>
    <div class="estimation-container__payout-area__right-area">
      <p class="title-text">$12000000.00</p>
    </div>
  </div>
  <!---->
</div>
`;

exports[`EstimationArea renders correctly with actual values and historical period with paystub 1`] = `
<div class="estimation-container">
  <div class="estimation-container__header">
    <p class="estimation-container__header__title">Info &amp; Estimation, <span class="estimation-container__header__period">Jan 2020</span></p>
    <div class="estimation-container__header__selection-area"><button name="Select Current Period" type="button" class="estimation-container__header__selection-area__item">
        <p class="estimation-container__header__selection-area__item__label long-text">
          Current Period
        </p>
        <p class="estimation-container__header__selection-area__item__label short-text">
          Current Per.
        </p>
      </button>
      <estimationperioddropdown-stub class="estimation-container__header__selection-area__item active"></estimationperioddropdown-stub>
    </div>
  </div>
  <div class="estimation-container__divider"></div>
  <div class="estimation-table-container">
    <div class="estimation-table-container__labels-area">
      <div class="column justify-start column-1">
        <p class="estimation-table-container__labels-area__text">Name</p>
      </div>
      <div class="column justify-start column-2">
        <p class="estimation-table-container__labels-area__text">Type</p>
      </div>
      <div class="column justify-start column-3">
        <p class="estimation-table-container__labels-area__text">Price</p>
      </div>
      <div class="column justify-start column-4">
        <p class="estimation-table-container__labels-area__text">Disk</p>
      </div>
      <div class="column justify-start column-5">
        <p class="estimation-table-container__labels-area__text">Bandwidth</p>
      </div>
      <div class="column justify-end column-6">
        <p class="estimation-table-container__labels-area__text">Payout</p>
      </div>
    </div>
    <div class="estimation-table-container__info-area">
      <div class="column justify-start column-1">
        <p class="estimation-table-container__info-area__text">Download</p>
      </div>
      <div class="column justify-start column-2">
        <p class="estimation-table-container__info-area__text">Egress</p>
      </div>
      <div class="column justify-start column-3">
        <p class="estimation-table-container__info-area__text">$20 / TB</p>
      </div>
      <div class="column justify-start column-4">
        <p class="estimation-table-container__info-area__text">--</p>
      </div>
      <div class="column justify-start column-5">
        <p class="estimation-table-container__info-area__text">0B</p>
      </div>
      <div class="column justify-end column-6">
        <p class="estimation-table-container__info-area__text">$0.00</p>
      </div>
    </div>
    <div class="estimation-table-container__info-area">
      <div class="column justify-start column-1">
        <p class="estimation-table-container__info-area__text">Repair &amp; Audit</p>
      </div>
      <div class="column justify-start column-2">
        <p class="estimation-table-container__info-area__text">Egress</p>
      </div>
      <div class="column justify-start column-3">
        <p class="estimation-table-container__info-area__text">$10 / TB</p>
      </div>
      <div class="column justify-start column-4">
        <p class="estimation-table-container__info-area__text">--</p>
      </div>
      <div class="column justify-start column-5">
        <p class="estimation-table-container__info-area__text">0B</p>
      </div>
      <div class="column justify-end column-6">
        <p class="estimation-table-container__info-area__text">$0.00</p>
      </div>
    </div>
    <div class="estimation-table-container__info-area">
      <div class="column justify-start column-1">
        <p class="estimation-table-container__info-area__text">Disk Average Month</p>
      </div>
      <div class="column justify-start column-2">
        <p class="estimation-table-container__info-area__text">Storage</p>
      </div>
      <div class="column justify-start column-3">
        <p class="estimation-table-container__info-area__text">$1.5 / TBm</p>
      </div>
      <div class="column justify-start column-4">
        <p class="estimation-table-container__info-area__text">0Bm</p>
      </div>
      <div class="column justify-start column-5">
        <p class="estimation-table-container__info-area__text">--</p>
      </div>
      <div class="column justify-end column-6">
        <p class="estimation-table-container__info-area__text">$0.00</p>
      </div>
    </div>
    <div class="estimation-table-container__info-area">
      <div class="column justify-start column-1">
        <p class="estimation-table-container__info-area__text">Gross Total</p>
      </div>
      <div class="column justify-start column-2"></div>
      <div class="column justify-start column-3"></div>
      <div class="column justify-start column-4"></div>
      <div class="column justify-start column-5"></div>
      <div class="column justify-end column-6">
        <p class="estimation-table-container__info-area__text">$0.44</p>
      </div>
    </div>
    <div class="estimation-table-container__total-area">
      <p class="estimation-table-container__total-area__text">Total + Surge (300%)</p>
      <p class="estimation-table-container__total-area__text">$1.33</p>
    </div>
    <div class="estimation-table-container__held-area">
      <p class="estimation-table-container__held-area__text">Held Back</p>
      <p class="estimation-table-container__held-area__text">-$0.78</p>
    </div>
    <!---->
    <div class="estimation-table-container__net-total-area">
      <div class="column justify-start column-1">
        <p class="estimation-table-container__net-total-area__text">NET TOTAL</p>
      </div>
      <div class="column justify-start column-2"></div>
      <div class="column justify-start column-3"></div>
      <div class="column justify-start column-4">
        <p class="estimation-table-container__net-total-area__text">0Bm</p>
      </div>
      <div class="column justify-start column-5">
        <p class="estimation-table-container__net-total-area__text">0B</p>
      </div>
      <div class="column justify-end column-6">
        <p class="estimation-table-container__net-total-area__text">$0.56</p>
      </div>
    </div>
    <div class="estimation-table-container__distributed-area">
      <div class="estimation-table-container__distributed-area__left-area">
        <p class="estimation-table-container__distributed-area__text">Distributed</p>
        <div class="estimation-table-container__distributed-area__info-area">
          <checksinfoicon-stub alt="Info icon with question mark" class="checks-area-image"></checksinfoicon-stub>
          <div class="tooltip" style="display: none;">
            <div class="tooltip__text-area">
              <p class="tooltip__text-area__text">If you see $0.00 as your distributed amount, you didn’t reach the minimum payout threshold. Your payout will be distributed along with one of the payouts in the upcoming payout cycles. If you see a distributed amount higher than expected, it means this month you were paid undistributed payouts from previous months in addition to this month’s payout.</p>
            </div>
            <div class="tooltip__footer"></div>
          </div>
        </div>
      </div>
      <p class="estimation-table-container__distributed-area__text">$0.33</p>
    </div>
  </div>
  <!---->
  <!---->
</div>
`;

exports[`EstimationArea renders correctly with actual values and historical period without paystub 1`] = `
<div class="estimation-container">
  <div class="estimation-container__header">
    <p class="estimation-container__header__title">Info &amp; Estimation, <span class="estimation-container__header__period">Jan 2020</span></p>
    <div class="estimation-container__header__selection-area"><button name="Select Current Period" type="button" class="estimation-container__header__selection-area__item">
        <p class="estimation-container__header__selection-area__item__label long-text">
          Current Period
        </p>
        <p class="estimation-container__header__selection-area__item__label short-text">
          Current Per.
        </p>
      </button>
      <estimationperioddropdown-stub class="estimation-container__header__selection-area__item active"></estimationperioddropdown-stub>
    </div>
  </div>
  <div class="estimation-container__divider"></div>
  <!---->
  <!---->
  <div class="no-data-container"><img src="@/../static/images/payments/NoData.png" class="no-data-container__image">
    <p class="no-data-container__title">No data to display</p>
    <p class="no-data-container__additional-text">Please note, historical data about payouts does not update immediately, it may take some time.</p>
  </div>
</div>
`;

exports[`EstimationArea renders correctly with actual values and previous period without paystub 1`] = `
<div class="estimation-container">
  <div class="estimation-container__header">
    <p class="estimation-container__header__title">Info &amp; Estimation, <span class="estimation-container__header__period">Jan 2020</span></p>
    <div class="estimation-container__header__selection-area"><button name="Select Current Period" type="button" class="estimation-container__header__selection-area__item">
        <p class="estimation-container__header__selection-area__item__label long-text">
          Current Period
        </p>
        <p class="estimation-container__header__selection-area__item__label short-text">
          Current Per.
        </p>
      </button>
      <estimationperioddropdown-stub class="estimation-container__header__selection-area__item active"></estimationperioddropdown-stub>
    </div>
  </div>
  <div class="estimation-container__divider"></div>
  <div class="estimation-table-container">
    <div class="estimation-table-container__labels-area">
      <div class="column justify-start column-1">
        <p class="estimation-table-container__labels-area__text">Name</p>
      </div>
      <div class="column justify-start column-2">
        <p class="estimation-table-container__labels-area__text">Type</p>
      </div>
      <div class="column justify-start column-3">
        <p class="estimation-table-container__labels-area__text">Price</p>
      </div>
      <div class="column justify-start column-4">
        <p class="estimation-table-container__labels-area__text">Disk</p>
      </div>
      <div class="column justify-start column-5">
        <p class="estimation-table-container__labels-area__text">Bandwidth</p>
      </div>
      <div class="column justify-end column-6">
        <p class="estimation-table-container__labels-area__text">Payout</p>
      </div>
    </div>
    <div class="estimation-table-container__info-area">
      <div class="column justify-start column-1">
        <p class="estimation-table-container__info-area__text">Download</p>
      </div>
      <div class="column justify-start column-2">
        <p class="estimation-table-container__info-area__text">Egress</p>
      </div>
      <div class="column justify-start column-3">
        <p class="estimation-table-container__info-area__text">$20 / TB</p>
      </div>
      <div class="column justify-start column-4">
        <p class="estimation-table-container__info-area__text">--</p>
      </div>
      <div class="column justify-start column-5">
        <p class="estimation-table-container__info-area__text">100MB</p>
      </div>
      <div class="column justify-end column-6">
        <p class="estimation-table-container__info-area__text">$2000000.00</p>
      </div>
    </div>
    <div class="estimation-table-container__info-area">
      <div class="column justify-start column-1">
        <p class="estimation-table-container__info-area__text">Repair &amp; Audit</p>
      </div>
      <div class="column justify-start column-2">
        <p class="estimation-table-container__info-area__text">Egress</p>
      </div>
      <div class="column justify-start column-3">
        <p class="estimation-table-container__info-area__text">$10 / TB</p>
      </div>
      <div class="column justify-start column-4">
        <p class="estimation-table-container__info-area__text">--</p>
      </div>
      <div class="column justify-start column-5">
        <p class="estimation-table-container__info-area__text">300MB</p>
      </div>
      <div class="column justify-end column-6">
        <p class="estimation-table-container__info-area__text">$4000000.00</p>
      </div>
    </div>
    <div class="estimation-table-container__info-area">
      <div class="column justify-start column-1">
        <p class="estimation-table-container__info-area__text">Disk Average Month</p>
      </div>
      <div class="column justify-start column-2">
        <p class="estimation-table-container__info-area__text">Storage</p>
      </div>
      <div class="column justify-start column-3">
        <p class="estimation-table-container__info-area__text">$1.5 / TBm</p>
      </div>
      <div class="column justify-start column-4">
        <p class="estimation-table-container__info-area__text">500MBm</p>
      </div>
      <div class="column justify-start column-5">
        <p class="estimation-table-container__info-area__text">--</p>
      </div>
      <div class="column justify-end column-6">
        <p class="estimation-table-container__info-area__text">$6000000.00</p>
      </div>
    </div>
    <div class="estimation-table-container__info-area">
      <div class="column justify-start column-1">
        <p class="estimation-table-container__info-area__text">Gross Total</p>
      </div>
      <div class="column justify-start column-2"></div>
      <div class="column justify-start column-3"></div>
      <div class="column justify-start column-4"></div>
      <div class="column justify-start column-5"></div>
      <div class="column justify-end column-6">
        <p class="estimation-table-container__info-area__text">$17000000.00</p>
      </div>
    </div>
    <!---->
    <div class="estimation-table-container__held-area">
      <p class="estimation-table-container__held-area__text">Held Back</p>
      <p class="estimation-table-container__held-area__text">-$9000000.00</p>
    </div>
    <!---->
    <div class="estimation-table-container__net-total-area">
      <div class="column justify-start column-1">
        <p class="estimation-table-container__net-total-area__text">NET TOTAL</p>
      </div>
      <div class="column justify-start column-2"></div>
      <div class="column justify-start column-3"></div>
      <div class="column justify-start column-4">
        <p class="estimation-table-container__net-total-area__text">500MBm</p>
      </div>
      <div class="column justify-start column-5">
        <p class="estimation-table-container__net-total-area__text">400MB</p>
      </div>
      <div class="column justify-end column-6">
        <p class="estimation-table-container__net-total-area__text">$8000000.00</p>
      </div>
    </div>
    <!---->
  </div>
  <!---->
  <!---->
</div>
`;
